src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -643,9 +673,26 @@ public Map<FingerprintIdDetails, Fingerprint> fingerprintFiles(final Collection<
                 log.debug("Processed file: " + p.toUri().toString() + " (" + filesRead.get() + ")");
 
                 if (CheckFingerprint.fileContainsReads(p)) {
-                    oneFileFingerprints = fingerprintSamFile(p, HaplotypeProbabilitiesFromSequence::new);
+                    if (indexPathMap != null && indexPathMap.containsKey(p)) {
```

Comment:
You can pull the checking for the index outside of the if statement.  for example 
```java
final Path indexPath = indexPathMap == null ? null : indexPathMap.get(p);
                if (indexPathMap != null && indexPath == null) {
                    log.warn("Index map file provided, but no explicit index provided for " + p);
                } else if (indexPath != null) {
                    log.info("Using explicit index provided for " + p);
                }
```
and then you can use `indexPath` in the call to fingerprintSamFile or fingerprintVcf

Reply:
Good idea, thanks! I refactored a bit to clean up this block of code.